### PR TITLE
Accessible listbox

### PR DESCRIPTION
**Pull request by seangates**
Covers #38

The new listbox styles are updated to a new `dropdown.css` which includes shared styles with menu and listbox.

As an added note: the listbox is actually a custom-styled "combobox".

### CODE REVIEW

File: dist/dropdown/ds6/dropdown.css [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+}
+.listbox {
+  display: inline-block;
+  box-sizing: border-box;
+  width: 100%;
+}
+.listbox:hover > .listbox-btn > input.listbox-btn__control {
+  border-color: #4295ff;
+}
+.listbox > .listbox-btn > input.listbox-btn__control {
```
Comment (ianmcburnie) [inline dist/dropdown/ds6/dropdown.css]: Do you think there's worth in changing `.listbox-btn` from a BEM block to a BEM element? Then specificity is reduced in quite a few places.

For example:

`.listbox > .listbox-btn {`

Becomes just

`.listbox__control {` (or some other name)

Or do you think `listbox-btn` is usable elsewhere? Like we do with `.expand-btn`? If so, we'll keep it as is.

Comment (seangates) [reply dist/dropdown/ds6/dropdown.css]: What happens to the input?:
```
.listbox__control > input { ... }
```
With no class for the nested input?

Comment (ianmcburnie) [reply dist/dropdown/ds6/dropdown.css]: That works.

Comment (seangates) [reply dist/dropdown/ds6/dropdown.css]: Just confirming, `listbox-btn` goes away in favor of `listbox__control`.

Comment (seangates) [reply dist/dropdown/ds6/dropdown.css]: Okay. This does mean more changes to my component code. [sigh]

Small price for improvements. 😉

**Comment by seangates** [comment]
@ianmcburnie Updated. Please continue your review.

File: src/less/listbox/ds6/listbox-base.less [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    display: inline-block;
+    box-sizing: border-box;
+    width: 100%;
+
+    &:hover > .listbox__control > input {
+        border-color: @ds6-color-p004-blue;
+    }
+}
+
+.listbox > .listbox__control > input {
```
Comment (ianmcburnie) [inline src/less/listbox/ds6/listbox-base.less]: I think this should be just `.listbox__control` now? And the same in all other occurences of `.listbox > .listbox__control`.

Comment (ianmcburnie) [reply src/less/listbox/ds6/listbox-base.less]: As you are out today, I can take care of this change and get this merged.

**Comment by seangates** [comment]
I’ll have a look.

Also, sorry about not removing the `.listbox` from the selectors. It was late after an evening of travel. 🤦🏻‍♂️😴

**Comment by ianmcburnie** [comment]
Don't worry about it. We're a team *fist bump*

**Comment by seangates** [comment]
Fixed locally, but I'm not happy about the font's not lining up. I've had issues like this before, though, where the `<input>` font and other inline fonts just don't want to play nice together.

screen shot 2018-02-02 at 11 10 35 pm

screen shot 2018-02-02 at 11 10 35 pm copy

Thoughts on the worth of fixing this right now, @ianmcburnie? I vote we create an issue to track, and prioritize it against other work.

**Pull request merged**